#! /usr/bin/env python
PACKAGE='traffic_color'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name          Type   Level 		Description              Default   Min    Max
gen.add("r1_h_h",    int_t, 0,       "Hue threshold center",   1,         0,    255)
gen.add("r1_h_l",    int_t, 0,       "Hue threshold width",    0,         0,    255)
gen.add("r1_s_h",    int_t, 0,       "Saturation threshold",   255,       0,    255)
gen.add("r1_s_l",    int_t, 0,       "Value threshold",        254,       0,    255)
gen.add("r1_v_h",    int_t, 0,       "Value threshold",        255,       0,    255)
gen.add("r1_v_l",    int_t, 0,       "Hue threshold center",   105,       0,    255)

gen.add("r2_h_h",    int_t, 0,       "Hue threshold center",   255,       0,    255)
gen.add("r2_h_l",    int_t, 0,       "Hue threshold width",    255,       0,    255)
gen.add("r2_s_h",    int_t, 0,       "Saturation threshold",   255,       0,    255)
gen.add("r2_s_l",    int_t, 0,       "Value threshold",        255,       0,    255)
gen.add("r2_v_h",    int_t, 0,       "Value threshold",        255,       0,    255)
gen.add("r2_v_l",    int_t, 0,       "Hue threshold center",   255,       0,    255)

gen.add("y1_h_h",    int_t, 0,       "Hue threshold center",   35,        0,    255)
gen.add("y1_h_l",    int_t, 0,       "Hue threshold width",    10,        0,    255)
gen.add("y1_s_h",    int_t, 0,       "Saturation threshold",   255,       0,    255)
gen.add("y1_s_l",    int_t, 0,       "Value threshold",        0,         0,    255)
gen.add("y1_v_h",    int_t, 0,       "Value threshold",        255,       0,    255)
gen.add("y1_v_l",    int_t, 0,       "Hue threshold center",   255,       0,    255)

gen.add("g1_h_h",    int_t, 0,       "Hue threshold center",   60,        0,    255)
gen.add("g1_h_l",    int_t, 0,       "Hue threshold width",    60,        0,    255)
gen.add("g1_s_h",    int_t, 0,       "Saturation threshold",   255,       0,    255)
gen.add("g1_s_l",    int_t, 0,       "Value threshold",        255,       0,    255)
gen.add("g1_v_h",    int_t, 0,       "Value threshold",        255,       0,    255)
gen.add("g1_v_l",    int_t, 0,       "Hue threshold center",   103,       0,    255)

exit(gen.generate(PACKAGE, PACKAGE, "traffic_color"))

